src/vsr/client.zig

```diff
@@ -382,18 +389,13 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
             assert(!self.request_queue.full());
             self.request_queue.push_assume_capacity(.{
                 .message = message_request,
-                .demux = .{
-                    .user_data = undefined, // Set during batch_submit().
-                    .callback = undefined, // Set during batch_submit().
-                    .event_count = @intCast(event_count),
-                    .event_offset = 0,
-                },
             });
 
             const request = self.request_queue.tail_ptr().?;
+            request.demux_queue.push(demux);
```

Comment:
Assert that `demux_queue.count == 0` here.
Also, we can assert that `demux_queue.count >1` above when appending a demux into an existing request.